bindings/c/test/fdb_api.hpp

```diff
@@ -556,6 +559,58 @@ class Database {
 	}
 };
 
+class Tenant final {
+	static const std::string tenantManagementMapPrefix;
+	static const std::string tenantMapPrefix;
+
+public:
+	static TypedFuture<future_var::None> createTenant(Transaction tr, std::string name) {
```

Comment:
Please use BytesRef for all string-like parameters, consistent with the rest of the API.

Reply:
The string manipulation becomes quite a hassle to deal with using the primitive types. Is there a good practice to follow for concatenating a BytesRef with the `constexpr std::string_view` for the tenant map prefix? I'm wondering how strict this requirement is, considering the `Database` constructor is accepting a `const std::string&` as the cluster file path parameter

Reply:
I think it is more about a design consistency than a strict requirement. We definitely want to avoid using std::string for operations on the performance-critical path, but createTenant is not performance critical. If you want to keep using string, then please pass parameters as const std::string&.

Reply:
You can use fmt::format("{}{}", prefix, str)

Reply:
The clearest guideline would be to use BytesRef for where C API takes uint8_t* and int length, std::string for when a null-terminated string is needed, and std::string_view for other cases.

The reason why Database Ctor takes std::string is because unlike key/value APIs, it takes const char* and expects it to be null-terminated. Only std::string::c_str() can guarantee that, not std::string_view.

Note that std::string_view has an implicit constructor that takes std::string: i.e. any function accepting string_view can implicitly accept string. Same goes for BytesRef (std::basic_string_view<uint8_t>) and ByteString (std::basic_string<uint8_t>).

Reply:
Note 2: fmt::format() will complain if you use uint8_t. In such case, wrap it in toCharsRef().

Reply:
Thank you for this information, this was very helpful to know